### PR TITLE
fix: 🐛 fiori theme updates for multiple components

### PR DESCRIPTION
**Pull request by xiaoyu0722**
HCPSDKFIORIUIKIT-2245 HCPSDKFIORIUIKIT-2257 HCPSDKFIORIUIKIT-2258 
HCPSDKFIORIUIKIT-2262 HCPSDKFIORIUIKIT-2264

### CODE REVIEW

File: Sources/FioriSwiftUICore/Views/SearchableListContent.swift [hunk 1]
```diff
@@ -96,7 +96,7 @@ struct SearchableListContent<Data: RandomAccessCollection, ID: Hashable, RowCont
                     UITableView.appearance().backgroundColor = .clear
                 }
                 .onDisappear {
-                    UITableView.appearance().backgroundColor = .systemGroupedBackground
+                    UITableView.appearance().backgroundColor = UIColor.preferredColor(.secondaryBackground)
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/Views/SearchableListContent.swift:99]: Use `UIColor.preferredFioriColor`

File: Sources/FioriThemeManager/Colors/UIColor+Extension.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+public extension UIColor {
+    /// Extension to `UIColor`, to return colors from the latest version of preset color palette used by `ThemeManager`.
+    ///
+    /// - parameters:
+    ///     - style: `ColorStyle` enum from the color palette.
+    ///     - scheme: specifies whether the color should be used in front of a *light* background, or a *dark* background.  E.g. a "white" background is a "light" background.  A "dark blue" background is a "dark" background.  Defaults to `.device`.
+    ///     - level: specifies whether the color sould be used in the *base* or *elevated* level of the interface. E.g. alerts and popovers will be assigned with the *elevated* interface level. Defaults to `.device`.
+    /// - Returns: a dynamic color provider wrapped in `UIColor`
+    static func preferredColor(_ style: ColorStyle, background scheme: BackgroundColorScheme? = .device, interface level: InterfaceLevel? = .device, display mode: ColorDisplayMode? = .device) -> UIColor {
```
Comment (billzhou0223) [inline Sources/FioriThemeManager/Colors/UIColor+Extension.swift:12]: Can we make it internal?

Rename to `preferredFioriColor`

File: Sources/FioriThemeManager/72-Fonts/UIFont+Fiori.swift [hunk 3]
```diff
@@ -0,0 +1,132 @@
+import Foundation
+import UIKit
+
+public extension UIFont {
```
Comment (billzhou0223) [inline Sources/FioriThemeManager/72-Fonts/UIFont+Fiori.swift:4]: Sheng already add this in #525.

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

File: Sources/FioriThemeManager/Colors/UIColor+Extension.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+import UIKit
+
+public extension UIColor {
```
[the remark this answers is not in the document]
Comment (xiaoyu0722) [reply Sources/FioriThemeManager/Colors/UIColor+Extension.swift:4]: But, `ThemeManager.shared.uiColor` is internal in `FioriThemeManager` module.
I need a public api, so I can call it in `FioriSwiftUICore`.

File: Sources/FioriSwiftUICore/Views/SearchableListContent.swift [hunk 5]
```diff
@@ -96,7 +96,7 @@ struct SearchableListContent<Data: RandomAccessCollection, ID: Hashable, RowCont
                     UITableView.appearance().backgroundColor = .clear
                 }
                 .onDisappear {
-                    UITableView.appearance().backgroundColor = .systemGroupedBackground
+                    UITableView.appearance().backgroundColor = Color.preferredColor(.secondaryBackground).uiColor()
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/Views/SearchableListContent.swift:99]: Not sure if color created like this is dynamic color.

Another option is to use `Color.preferredUIColor()`

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

File: Sources/FioriSwiftUICore/Views/SearchableListContent.swift (hunk 5 above)
Comment (xiaoyu0722) [reply Sources/FioriSwiftUICore/Views/SearchableListContent.swift:99]: Yes, its dynamic. `preferredColor` and `preferredUIColor` are using same implementation.
``` swift
internal func color(for style: ColorStyle, background scheme: BackgroundColorScheme?, interface level: InterfaceLevel?, display mode: ColorDisplayMode?) -> Color {
    let uiColor = self.uiColor(for: style, background: scheme, interface: level, display: mode)
    let color = Color(uiColor)
    return color
}

**Review by billzhou0223 — APPROVED** [review]

**Comment by xiaoyu0722** [comment]
All commits in #530

**Pull request closed without merging**